packages/react/src/components/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,379 @@
+import React, { CSSProperties, FC, PropsWithChildren, ReactNode, useState } from 'react';
+import { useCombobox, useMultipleSelection, useSelect } from 'downshift';
+
+import styles from './Dropdown.module.css';
+import classNames from '../../utils/classNames';
+import { IconAngleDown, IconCheck, IconAttention } from '../../icons';
+import Checkbox from '../checkbox/Checkbox';
+
+type OptionType = {
+  [key: string]: any;
+};
+
+export type DropdownProps = {
+  /**
+   * Controls the circular keyboard navigation between items. If set to `true`, when first item is highlighted, the Arrow Up will move highlight to the last item, and vice versa using Arrow Down.
+   */
+  circularNavigation?: boolean;
+  /**
+   * Additional class names to apply to the dropdown
+   */
+  className?: string;
+  /**
+   * Close the menu when the user selects an option
+   */
+  closeMenuOnSelect?: boolean;
+  /**
+   * If `true`, the dropdown will be disabled
+   */
+  disabled?: boolean;
+  /**
+   * If `true`, the dropdown menu can be filtered
+   */
+  filterable?: boolean;
+  /**
+   * Function used to get `aria` attributes and the `id` prop for menu options (`li`)
+   * @param index Data item index
+   */
+  getItemId?: (index: number) => string;
```

Comment:
This gets me confused. This return a single `id` as string, right? I think the `aria` part here is misleading, at least to me.